### PR TITLE
Add RC draft post command, make some minor improvements to beta post command

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

As a follow up to #37142 this adds a very similar command to make the RC post. Like the beta post I've kept it very simple for now to avoid complications.

The goal is to avoid a lot of the copy/pasting involved in creating a new post and get you very close to a complete post. Later we can enhance this to add the change list.

I also back ported some minor improvements to output and documentation to the beta post generator (such as outputting an edit URL to save you some more time.) And fixing a bug where we didn't validate that the version passed has the correct type (e.g. running `rc` command you cannot pass a beta version)

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. First you'll need to check out this branch and ensure that `pnpm i` has been run on the monorepo
2. You'll also need an oauth app set up on developer.wordpress.com. See [the Readme for instructions](https://github.com/woocommerce/woocommerce/blob/e43bdcca60c44c87f9ef0fa9000a7424355b368a/tools/release-posts/README.md?plain=1#L17) and make sure as part of that you set the secrets required in a local `.env` file.
2. Traverse to the release post tool: `cd tools/release-posts`
3. Now run a test command for a fictitious RC post (don't worry even if you don't pass `--outputOnly` it will only post a draft
4. example command: `pnpm release-post rc 7.5.0-rc.1 --outputOnly`. See if you like the output
5. now to get a better idea of how the output looks, post a draft: `pnpm release-post rc 7.5.0-rc.1`
6. you can also test that it rejects a beta to rc command and rejects a rc to beta command. e.g. this should produce an error: `pnpm release-post rc 7.5.0-beta.1`

Note that during the command it will show you a list of posts it searched to link the last release post. I don't expect this to be perfect and it will sometimes not find something suitable but the idea is best effort and you can edit it manually later if it didn't find a good post.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: tools/release-posts/templates/rc-release.ejs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:heading {"anchor":"testing"} -->
+<h2 class="wp-block-heading" id="testing">Testing</h2>
+<!-- /wp:heading -->
+
+<!-- wp:paragraph -->
+<p>
+	If you'd like to dive in and help test this new release, our handy&nbsp;<a
```
Comment (psealock) [inline tools/release-posts/templates/rc-release.ejs]: There's a space before the "If"

image

Comment (psealock) [reply tools/release-posts/templates/rc-release.ejs]: Looking further, every paragraph beginning in the "Testing", "What's New?" and "Release Schedule" has an extra space too.

File: tools/release-posts/lib/github-api.ts [hunk 2]
```diff
@@ -115,3 +115,31 @@ export const getMostRecentFinal = async () => {
...
+
+	if ( betaReleases.length === 0 ) {
+		throw new Error( 'No beta releases found' );
+	}
+
+	const latestBetaRelease = betaReleases.reduce( ( latest, current ) => {
+		const latestDate = new Date( latest.created_at );
+		const currentDate = new Date( current.created_at );
+		return currentDate > latestDate ? current : latest;
+	} );
```
Comment (psealock) [inline tools/release-posts/lib/github-api.ts:142]: I do love a good `reduce` function

**Review by psealock — DISMISSED** [review]
This is testing well for me, just a few errant spaces but otherwise code looks good too.

**Comment by samueljseay** [comment]
Thanks @psealock that should be fixed now, could you take another look?

**Review by psealock — APPROVED** [review]
Changes look good 🚢

**Pull request merged**